src/zenml/utils/analytics_utils.py

```diff
@@ -333,11 +333,15 @@ def track(
         # infinite loop
         if gc._zen_store is not None:
             zen_store = gc.zen_store
+            user = zen_store.get_user()
+
+            if "user_id" not in properties:
+                properties["user_id"] = str(user.id)
+
             if (
                 zen_store.type == StoreType.REST
                 and "server_id" not in properties
             ):
-                user = zen_store.get_user()
                 server_info = zen_store.get_store_info()
                 properties.update(
```

Comment:
Can we have `client_id: self.user_id` in here please (after the discussion today)

Reply:
Done

Reply:
@schustmi yes, he just wants it duplicated as an event metadata field, to help with grouping and filtering (I think).